frontend/src/context/RidesContext.tsx

```diff
@@ -23,23 +24,101 @@ type RidesProviderProps = {
   children: React.ReactNode;
 };
 
+function uuidv4() {
+  return '10000000-1000-4000-8000-100000000000'.replace(/[018]/g, (c) =>
+    (
+      +c ^
+      (crypto.getRandomValues(new Uint8Array(1))[0] & (15 >> (+c / 4)))
+    ).toString(16)
+  );
+}
+
 export const RidesProvider = ({ children }: RidesProviderProps) => {
   const [unscheduledRides, setUnscheduledRides] = useState<Ride[]>([]);
   const [scheduledRides, setScheduledRides] = useState<Ride[]>([]);
   const { curDate } = useDate();
   const date = format_date(curDate);
 
   const refreshRides = useCallback(async () => {
-    const ridesData: Ride[] = await axios
+    const ridesDataToday: Ride[] = await axios
       .get(`/api/rides?date=${date}`)
       .then((res) => res.data)
       .then((data) => data.data);
-    if (ridesData) {
+
+    const newRecurringRides: Ride[] = await axios
```

Comment:
I believe the API is returning undefined data to ridesDataToday. An idea for error handling is to use a try catch to log what data you are fetching and see if it expected or not.